src/Engine/Audio/VorbisFile.cpp

```diff
@@ -1,28 +1,38 @@
 #include "VorbisFile.hpp"
 
-#include <Utility/Log.hpp>
+#include <algorithm>
 #include <stb_vorbis.c>
+#include <Utility/Log.hpp>
 
 #ifdef USINGMEMTRACK
 #include <MemTrackInclude.hpp>
 #endif
 
 using namespace Audio;
 
-Audio::VorbisFile::VorbisFile(){
+VorbisFile::VorbisFile() {
 
 }
 
 VorbisFile::~VorbisFile() {
-    stb_vorbis_close(stbFile);
+    if (stbFile)
+        stb_vorbis_close(stbFile);
+
+    if (buffer)
+        delete[] buffer;
 }
 
-int VorbisFile::GetData(uint32_t offset, uint32_t samples, float* data) const {
-    if (stbFile == nullptr)
+int VorbisFile::GetData(uint32_t offset, uint32_t samples, float*& data) const {
+    if (!IsLoaded())
         return 0;
 
-    stb_vorbis_seek(stbFile, offset);
-    return stb_vorbis_get_samples_float_interleaved(stbFile, channelCount, data, samples);
+    if (buffer) {
+        data = &buffer[offset];
+        return std::min(std::max(sampleCount - (int)(offset + samples), 0), (int)samples);
+    } else {
+        stb_vorbis_seek(stbFile, offset);
+        return stb_vorbis_get_samples_float_interleaved(stbFile, channelCount, data, samples);
+    }
 }
 
 uint32_t VorbisFile::GetSampleCount() const {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') The function 'GetSampleCount' is never used. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)